lib/twitter/rest/request.rb

```diff
@@ -108,7 +115,14 @@ def symbolize_keys!(object)
 
       # @return [HTTP::Client, HTTP]
       def http_client
-        @client.proxy ? HTTP.via(*proxy) : HTTP
+        client_with_timeout(@client.proxy ? HTTP.via(*proxy) : HTTP)
```

Comment:
Any call to `http_client` will call `client_with_timeout` so I'd be ok with just having the timeout options set here and removing the method. Pretty trivial though so feel free to ignore this.